routing/payment_lifecycle.go

```diff
@@ -808,7 +810,16 @@ func (p *shardHandler) handleFailureMessage(rt *route.Route,
 		return err
 	}
 
-	// Apply channel update.
+	// Apply channel update to additional edges first. If no update is
+	// performed, we will continue to update the channel edge policy in our
+	// db.
+	if p.paySession != nil && p.paySession.UpdateAdditionalEdge(
```

Comment:
Alternatively, couldn't we check that the chan ID being updated is amongst the set of ephemeral edges?

Reply:
This is changed, now we first lookup the edge policy inside `additionalEdges`, then decide which one to update.